# HG changeset patch
# User Jeff A. Marr <jamarr@gleim.com
# Date 1373554934 14400
# Node ID a016c516cec6419fd67091f64ce124dadd3cf044
# Parent  fe09e5f952c482669eb2a57b12fda1fc0c02183d
[1] re-applied patch for to wxWidgets 2.9.4 adding wxHtmlWindow::SetDefaultHTMLCursor.

diff -r fe09e5f952c4 -r a016c516cec6 wxWidgets/include/wx/html/htmlwin.h
--- a/wxWidgets/include/wx/html/htmlwin.h	Thu Jul 11 09:50:50 2013 -0400
+++ b/wxWidgets/include/wx/html/htmlwin.h	Thu Jul 11 11:02:14 2013 -0400
@@ -388,6 +388,7 @@
 
     /// Returns standard HTML cursor as used by wxHtmlWindow
     static wxCursor GetDefaultHTMLCursor(HTMLCursor type);
+	static void SetDefaultHTMLCursor(HTMLCursor type, const wxCursor& cursor);
 
 protected:
     void Init();
@@ -553,6 +554,7 @@
     // standard mouse cursors
     static wxCursor *ms_cursorLink;
     static wxCursor *ms_cursorText;
+	static wxCursor *ms_cursorDefault;
 
     DECLARE_EVENT_TABLE()
     wxDECLARE_NO_COPY_CLASS(wxHtmlWindow);
diff -r fe09e5f952c4 -r a016c516cec6 wxWidgets/src/html/htmlwin.cpp
--- a/wxWidgets/src/html/htmlwin.cpp	Thu Jul 11 09:50:50 2013 -0400
+++ b/wxWidgets/src/html/htmlwin.cpp	Thu Jul 11 11:02:14 2013 -0400
@@ -279,6 +279,7 @@
 wxHtmlProcessorList *wxHtmlWindow::m_GlobalProcessors = NULL;
 wxCursor *wxHtmlWindow::ms_cursorLink = NULL;
 wxCursor *wxHtmlWindow::ms_cursorText = NULL;
+wxCursor *wxHtmlWindow::ms_cursorDefault = NULL;
 
 void wxHtmlWindow::CleanUpStatics()
 {
@@ -289,6 +290,7 @@
     wxDELETE(m_GlobalProcessors);
     wxDELETE(ms_cursorLink);
     wxDELETE(ms_cursorText);
+	wxDELETE(ms_cursorDefault);
 }
 
 void wxHtmlWindow::Init()
@@ -1783,7 +1785,9 @@
 
         case HTMLCursor_Default:
         default:
-            return *wxSTANDARD_CURSOR;
+            if ( !ms_cursorDefault )
+                ms_cursorDefault = new wxCursor(wxCURSOR_ARROW);
+            return *ms_cursorDefault;
     }
 }
 
@@ -1792,6 +1796,30 @@
     return GetDefaultHTMLCursor(type);
 }
 
+/*static*/
+void wxHtmlWindow::SetDefaultHTMLCursor(HTMLCursor type, const wxCursor& cursor)
+{
+    switch (type)
+    {
+        case HTMLCursor_Link:
+            if ( ms_cursorLink )
+                delete ms_cursorLink;
+            ms_cursorLink = new wxCursor(cursor);
+            return;
+
+        case HTMLCursor_Text:
+            if ( ms_cursorText )
+                delete ms_cursorText;
+            ms_cursorText = new wxCursor(cursor);
+            return;
+
+        case HTMLCursor_Default:
+        default:
+            if ( ms_cursorDefault )
+                delete ms_cursorText;
+            ms_cursorDefault = new wxCursor(cursor);
+    }
+}
 
 //-----------------------------------------------------------------------------
 // wxHtmlWinModule
